Assets/Scripts/MazeGenerator.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.Collections;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeGenerator : MonoBehaviour
+{
+    [SerializeField]
+    private int width;
+
+    [SerializeField]
+    private int height;
+
+    public Transform Brick;
+    private int[,] Maze;
+    private List<Vector3> pathMazes = new List<Vector3>();
+    private Stack<Vector2> _tiletoTry = new Stack<Vector2>();
+    private List<Vector2> offsets = new List<Vector2> { new Vector2(0, 1), new Vector2(0, -1), new Vector2(1, 0), new Vector2(-1, 0) };
+    private System.Random rnd = new System.Random();
+    private int _width, _height;
+    private Vector2 _currentTile;
+    public Vector2 CurrentTile
+    {
+        get
+        {
+            return _currentTile;
+        }
+
+        private set
+        {
+            if (value.x < 1 || value.x >= this.width - 1 || value.y < 1 || value.y >= this.height - 1)
+            {
+                throw new ArgumentException("CurrentTile must be within the one tile border all around the maze");
+            }
+            if (value.x % 2 == 1 || value.y % 2 == 1)
+            {
+                _currentTile = value;
+            }
+            else
+            {
+                throw new ArgumentException("The current square must not be both on an even X-axis and an even Y-axis, to ensure we can get walls around all tunnels");
+            }
+        }
+    }
+    public static MazeGenerator Instance { get; private set; }
+
+    private void Awake()
+    {
+        Instance = this;
+        GenerateMaze();
+    }
+
+    private void Start()
+    {
+
+    }
+
+    private void GenerateMaze()
+    {
+        Maze = new int[width, height];
+        for (int x = 0; x < width; x++)
+        {
+            for (int y = 0; y < height; y++)
+            {
+                Maze[x, y] = 1;
+            }
+        }
+        CurrentTile = Vector2.one;
+        _tiletoTry.Push(CurrentTile);
+        Maze = CreateMaze();
+        //GameObject ptype = null;
+
+        for (int i = 0; i <= Maze.GetUpperBound(0); i++)
+        {
+            for (int j = 0; j <= Maze.GetUpperBound(0); j++)
+            {
+                if (Maze[i, j] == 1)
+                {
+                    //ptype = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                    //ptype.transform.position = new Vector3(i * ptype.transform.localScale.x, j * ptype.transform.localScale.y, 0);
+                    //if (brick != null)
+                    //{
+                    //    ptype.renderer.material = brick;
+                    //}
+                    //ptype.transform.parent = transform;
+
+                    Instantiate(Brick, new Vector3(i, 0, j), Quaternion.identity);
+                }
+                else if (Maze[i, j] == 0)
+                {
+                    pathMazes.Add(new Vector3(i, 0, j));
+                }
+            }
+        }
+    }
+
+    public int[,] CreateMaze()
```

Comment:
this should be a private as it isn't used anywhere else.

Reply:
Also, what is the value of having it return the `int[,]` if it only updates the private variable?

Reply:
Also, you have overcommented this function. It would be better to name the function based on the algorithm used and then only provide comments for unique functions. We can look up the algorithm details and it will likely be more useful if someone needs to learn the specifics of the maze generation algorithm you have chosen